### PR TITLE
v3.2.0

### PR DESCRIPTION
**Pull request by peter-applyinnovations**
## Related To:

https://applyinnovations.atlassian.net/browse/FCMA-96
### Changes:

- [x] useAuth, to update cognitoUser on userAttribute update
- [x] useAuth, to include userAuth

### CODE REVIEW

File: packages/auth/src/types.ts [hunk 1]
```diff
@@ -1,20 +1,20 @@
-﻿import { CognitoUser } from 'amazon-cognito-identity-js';
+﻿import { CognitoUser, CognitoUserAttribute } from "amazon-cognito-identity-js";
 
 export type ProfileTypes =
```
Comment (applyinnovations) [inline packages/auth/src/types.ts]: remove profile types no longer needed

File: packages/auth/dist/src/types.d.ts [hunk 2]
```diff
@@ -28,4 +28,6 @@ export interface AuthContextValues<CustomUserAttributes = {}> {
     confirmSignUp: (phoneNumber: string, answer: string) => Promise<ConfirmationResult>;
     confirmSignIn: (answer: string) => Promise<ConfirmationResult>;
     signOutUser: () => Promise<void>;
+    updateUserData: (data: CognitoUserAttribute) => Promise<void>;
```
Comment (applyinnovations) [inline packages/auth/dist/src/types.d.ts]: change `updateUserData` to `updateUserAttributes`

File: packages/auth/src/useAuth.tsx [hunk 3]
```diff
@@ -111,6 +113,17 @@ export function authContextValues<CustomUserAttributes = any>({
     []
   );
 
+  const updateUserData = useCallback(
```
Comment (applyinnovations) [inline packages/auth/src/useAuth.tsx]: change `updateUserData` to `updateUserAttributes`

File: packages/auth/src/types.ts [hunk 4]
```diff
@@ -1,20 +1,20 @@
-﻿import { CognitoUser } from 'amazon-cognito-identity-js';
+﻿import { CognitoUser, CognitoUserAttribute } from "amazon-cognito-identity-js";
```
Comment (applyinnovations) [inline packages/auth/src/types.ts]: create your own type here

```
interface UserAttributes {
  address: string;
  birthdate: string;
  email: string;
  family_name: string;
  gender: string;
  given_name: string;
  locale: string;
  middle_name: string;
  name: string;
  nickname: string;
  phone_number: string;
  picture: string;
  preferred_username: string;
  profile: string;
  updated_at: string;
  website: string;
  zoneinfo: string;
  phone_number_verified: boolean;
  email_verified: boolean;
}
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**